backend/storage/src/repositories/friendship_repository.rs

```diff
@@ -0,0 +1,441 @@
+                    receiver.id as receiver_user_id,
+                    receiver.username as receiver_username,
+                    receiver.avatar as receiver_avatar,
+                    receiver.banned as receiver_banned,
+                    receiver.warned as receiver_warned
+                FROM friend_requests fr
+                JOIN users sender ON fr.sender_id = sender.id
+                JOIN users receiver ON fr.receiver_id = receiver.id
+                WHERE ((fr.sender_id = $1 AND fr.receiver_id = $2) OR (fr.sender_id = $2 AND fr.receiver_id = $1))
+                AND fr.status = 'pending'
+            "#,
+            user1_id,
+            user2_id
+        )
+        .fetch_optional(self.borrow())
+        .await?;
+
+        let pending_request_with_user = pending_request.map(|row| FriendRequestWithUser {
+            id: row.id,
+            sender: UserLiteAvatar {
+                id: row.sender_user_id,
+                username: row.sender_username,
+                avatar: row.sender_avatar,
+                banned: row.sender_banned,
+                warned: row.sender_warned,
+            },
+            receiver: UserLiteAvatar {
+                id: row.receiver_user_id,
+                username: row.receiver_username,
+                avatar: row.receiver_avatar,
+                banned: row.receiver_banned,
+                warned: row.receiver_warned,
+            },
+            status: row.status,
+            message: row.message,
+            created_at: row.created_at,
+            updated_at: row.updated_at,
+        });
+
+        Ok(FriendshipStatus {
+            are_friends: false,
+            pending_request: pending_request_with_user,
+        })
+    }
+
+    /// Remove a friendship
+    pub async fn remove_friendship(&self, user1_id: i64, user2_id: i64) -> Result<()> {
+        let min_id = std::cmp::min(user1_id, user2_id);
+        let max_id = std::cmp::max(user1_id, user2_id);
+
+        let result = sqlx::query!(
+            r#"
+                DELETE FROM friendships
+                WHERE user1_id = $1 AND user2_id = $2
+            "#,
+            min_id,
+            max_id
+        )
+        .execute(self.borrow())
+        .await?;
+
+        if result.rows_affected() == 0 {
+            return Err(Error::BadRequest("Friendship not found".to_string()));
+        }
+
+        Ok(())
+    }
+
+    /// Cancel a sent friend request
+    pub async fn cancel_friend_request(
+        &self,
+        sender_id: i64,
+        friend_request_id: i64,
+    ) -> Result<()> {
+        let result = sqlx::query!(
+            r#"
+                DELETE FROM friend_requests
+                WHERE id = $1 AND sender_id = $2 AND status = 'pending'
+            "#,
+            friend_request_id,
+            sender_id
+        )
+        .execute(self.borrow())
+        .await?;
+
+        if result.rows_affected() == 0 {
+            return Err(Error::BadRequest(
```

Comment:
this is not a bad request but you need to add a new error `Error::FriendshipNotFound` and add the variant here https://github.com/Arcadia-Solutions/arcadia/blob/22950938e96c89eed2fa5c65924123e9e8939c91/backend/common/src/error/mod.rs#L265

Bad request and Not found are different http status codes